client/dashboard/app/queries/site-jetpack-module.ts

```diff
@@ -23,12 +24,18 @@ export const siteJetpackModuleMutation = ( siteId: number ) =>
 				: deactivateJetpackModule( siteId, module );
 		},
 		onSuccess: ( newData: unknown, variables: { module: string; value: boolean } ) => {
-			queryClient.setQueryData( siteJetpackModulesQuery( siteId ).queryKey, ( oldData = [] ) => {
-				if ( variables.value ) {
-					return [ ...oldData, variables.module ];
+			queryClient.setQueryData(
+				siteJetpackModulesQuery( siteId ).queryKey,
+				( oldData: Record< string, JetpackModule > = {} ) => {
```

Comment:
The explicit type isn't needed here. We've found that using an explicit type param for `setQueryData` can actually cover up bugs, so it is better to just let the types flow naturally.


```suggestion
				( oldData = {} ) => {
```

Interestingly the default `{}` isn't needed anymore, since spreading an `undefined` into an object is totally fine.